src/middlewares/createInsightsMiddleware.ts

```diff
@@ -88,25 +90,29 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = (props) => {
     });
     insightsClient('init', { appId, apiKey, ...insightsInitParams });
 
+    const createWidget = connectConfigure(noop);
+    let widget: ReturnType<typeof createWidget>;
+
     return {
       onStateChange() {},
       subscribe() {
         insightsClient('addAlgoliaAgent', 'insights-middleware');
 
-        // At the time this middleware is subscribed, `mainIndex.init()` is already called.
-        // It means `mainIndex.getHelper()` exists.
-        const helper = instantSearchInstance.mainIndex.getHelper()!;
```

Comment:
The insights middleware used to rely on the helper to set/get user token. However, now that we get this `subscribe()` called before `mainIndex.init()` we no longer have access to it at subscribe(). We could setTimeout(fn, 0) but it will make the first query being fired without user token.

Anyway I don't like touching helper directly here, so changed the implementation to create a custom configure widget within the middleware to manipulate the search parameters.